implement the crud operations for carosel slides

rshgithub opened the pull request.

rshgithub committed:
implement the crud operations for carosel slides

rshgithub committed:
commit changes

rshgithub committed:
Merge branch 'backend-dev' into FPBG3-66-implement-CrouselSlides-Crud-Operations

rshgithub committed:
final commit

AhmadAlBarasy commented on src/controllers/carouselSlidesController.ts:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    await carouselSlide.destroy();
+    res.status(202).json({ status: 'success' });
```
status code should be `204 no content`

AhmadAlBarasy commented on src/validators/carouselSlideFieldsValidation.ts:
```diff
@@ -0,0 +1,125 @@
+    }),
+
+  title: Joi.string()
+    .max(100)
```
you used for the title `Datatypes.STRING` as a datatype, and it accepts up to 255 characters which leads to inconsistency, you either have to change teh value here to be 255 characters or to edit the size in the model to be `DataTypes.STRING(100)`

AhmadAlBarasy commented on src/validators/carouselSlideFieldsValidation.ts:
```diff
@@ -0,0 +1,125 @@
+    }),
+
+  categoryName: Joi.string()
+    .max(100)
```
the category name here should be 50 characters max. not 100 (because in our Category model, the limit is 50)

AhmadAlBarasy commented on src/validators/carouselSlideFieldsValidation.ts:
```diff
@@ -0,0 +1,125 @@
+    }),
+
+  brandName: Joi.string()
+    .max(100)
```
Same goes here, only 50 characters

AhmadAlBarasy commented on src/validators/carouselSlideFieldsValidation.ts:
```diff
@@ -0,0 +1,125 @@
+    }),
+
+  categoryName: Joi.string()
+    .max(100)
```
Same here, `max(50)`

AhmadAlBarasy commented on src/validators/carouselSlideFieldsValidation.ts:
```diff
@@ -0,0 +1,125 @@
+    }),
+
+  brandName: Joi.string()
+    .max(100)
```
And here also `max(50)`

AhmadAlBarasy commented on src/controllers/productsController.ts:
It's not a good practice at all to make changes on files that are not related to the purpose of opening the branch, please revert these changes and you can make them in a separate task with a separate branch.

AhmadAlBarasy reviewed (changes requested):
Please check the comments I left, and we must have a discussion tomorrow about the implementation of the CarouselSlide model because there are some points that are unclear to me

rshgithub committed:
final commit changes
- category or brand are optional for carosel slide
- fix status code
- edit the validation
- more readable code .

rshgithub committed:
Merge branch 'backend-dev' into FPBG3-66-implement-CrouselSlides-Crud-Operations
